src/pages/patientView/clinicalInformation/PatientViewPageStore.ts

```diff
@@ -1696,6 +1788,173 @@ export class PatientViewPageStore {
+            invoke: async () =>
+                fetchOncoKbDataForOncoprint(
+                    this.oncoKbAnnotatedGenes,
+                    this.mutationData
+                ),
+            onError: (err: Error) => {
+                // fail silently, leave the error handling responsibility to the data consumer
+            },
+        },
+        ONCOKB_DEFAULT
+    );
+
+    readonly cnaOncoKbDataForOncoprint = remoteData<IOncoKbData | Error>(
+        {
+            await: () => [
+                this.clinicalDataForSamples,
+                this.studiesForSamplesWithoutCancerTypeClinicalData,
+                this.samplesWithoutCancerTypeClinicalData,
+                this.oncoKbAnnotatedGenes,
+                this.molecularData,
+                this.molecularProfileIdToMolecularProfile,
+            ],
+            invoke: async () =>
+                fetchCnaOncoKbDataForOncoprint(
+                    this.oncoKbAnnotatedGenes,
+                    this.molecularData,
+                    this.molecularProfileIdToMolecularProfile
+                ),
+        },
+        ONCOKB_DEFAULT
+    );
+
+    readonly getPutativeDriverInfo = remoteData({
+        await: () => [
+            this.getOncoKbMutationAnnotationForOncoprint,
+            this.isHotspotForOncoprint,
+        ],
+        invoke: () => {
+            return Promise.resolve((mutation: Mutation): {
+                oncoKb: string;
+                hotspots: boolean;
+                cbioportalCount: boolean;
+                cosmicCount: boolean;
+                customDriverBinary: boolean;
+                customDriverTier?: string;
+            } => {
+                const getOncoKbMutationAnnotationForOncoprint = this
+                    .getOncoKbMutationAnnotationForOncoprint.result!;
+                const oncoKbDatum:
+                    | IndicatorQueryResp
+                    | undefined
+                    | null
+                    | false =
+                    getOncoKbMutationAnnotationForOncoprint &&
+                    !(
+                        getOncoKbMutationAnnotationForOncoprint instanceof Error
+                    ) &&
+                    getOncoKbMutationAnnotationForOncoprint(mutation);
+
+                let oncoKb: string = '';
+                if (oncoKbDatum) {
+                    oncoKb = getOncoKbOncogenic(oncoKbDatum);
+                }
+
+                const hotspots: boolean =
+                    !(this.isHotspotForOncoprint.result instanceof Error) &&
+                    this.isHotspotForOncoprint.result!(mutation);
+
+                return {
+                    oncoKb,
+                    hotspots,
+                    // TODO support these too?
+                    cbioportalCount: false,
+                    cosmicCount: false,
+                    customDriverBinary: false,
+                };
+            });
+        },
+    });
+
+    readonly getOncoKbMutationAnnotationForOncoprint = remoteData<
```

Comment:
it's probably too late but i don't think we should name these promises "get" prefix

Reply:
Right, we have a promise with the same name in the results view page store.

Reply:
the result of the remoteData is a function, so that's where the `get` comes from. I remember we had discussed this and you didn't like it. Im not crazy about it either but what I like about it is it combines a function with its asynchronous dependencies so that a caller only has to ask whether the function is ready, without having to reference its dependencies separately. @alisman